Update OP2MissionSDK to version 4.0.0

Brett208 opened the pull request.

Brett208 committed:
Update OP2MissionSDK to version 2.0.0

DanRStevens reviewed (commented):
Changes look good. Nice to get these updates in.

In terms of version numbers, I think we've had previous releases of the "Outpost 2 SDK" known as versions 1-3? I don't really remember. We weren't very good at tracking that stuff. We may want to give some thought to the version numbers to avoid ambiguity.

Edit:
This thread contains a number of references to version numbers:
https://forum.outpost2.net/index.php?topic=6018.0
In particular, there is already a `v2.0.0` tag for the `Outpost2DLL` project, and it from way back in 2008.
There was also mention of OP2Helper having a 2.1 tag, though I'm not seeing that in the repo. Maybe an oversight.

I think we had a zip file floating around somewhere that was tagged as V3.

Brett208 committed:
Change from version 2.0.0 to 4.0.0

Versions 1-3 existed on SVN repository and in forums

Brett208 committed:
Add change log for version 3.5.0

Version 3.5.0 was never released and is used for a placeholder for changes made since 2015 before switching from SVN to Git.

Brett208 committed:
Update version of OP2Helper

 - Fixes OP2Helper to not compile in debug mode when release configuration selected
 - General refactors of OP2Helper

DanRStevens reviewed (approved):
Changes look good.

Brett208 committed:
Add Outpost2App

Brett208 committed:
Remove BaseBuilderV2
